website/docs/guides/service_principal_oidc.html.markdown

```diff
@@ -0,0 +1,173 @@
+---
+layout: "azurerm"
+page_title: "Azure Provider: Authenticating via a Service Principal and Open ID Connect"
+description: |-
+  This guide will cover how to use a Service Principal (Shared Account) with Open ID Connect as authentication for the Azure Provider.
+---
+
+# Azure Provider: Authenticating using a Service Principal with Open ID Connect
+
+Terraform supports a number of different methods for authenticating to Azure:
+
+* [Authenticating to Azure using the Azure CLI](azure_cli.html)
+* [Authenticating to Azure using Managed Service Identity](managed_service_identity.html)
+* [Authenticating to Azure using a Service Principal and a Client Certificate](service_principal_client_certificate.html)
+* [Authenticating to Azure using a Service Principal and a Client Secret](service_principal_client_secret.html)
+* Authenticating to Azure using a Service Principal and Open ID Connect (which is covered in this guide)
+---
+
+We recommend using either a Service Principal or Managed Service Identity when running Terraform non-interactively (such as when running Terraform in a CI server) - and authenticating using the Azure CLI when running Terraform locally.
+
+## Setting up an Application and Service Principal
+
+A Service Principal is a security principal within Azure Active Directory which can be granted access to resources within Azure Subscriptions. To authenticate with a Service Principal, you will need to create an Application object within Azure Active Directory, which you will use as a means of authentication, either [using a Client Secret](service_principal_client_secret.html), [a Client Certificate](service_principal_client_certificate.html), or Open ID Connect (which is documented in this guide). This can be done using the Azure Portal.
```

Comment:
```suggestion
A Service Principal is a security principal within Azure Active Directory which can be granted access to resources within Azure Subscriptions. To authenticate with a Service Principal, you will need to create an Application object within Azure Active Directory, which you will use as a means of authentication, either [using a Client Secret](service_principal_client_secret.html), [a Client Certificate](service_principal_client_certificate.html), or OpenID Connect (which is documented in this guide). This can be done using the Azure Portal.
```